backend/ml/management/commands/ml_train.py

```diff
@@ -50,48 +50,72 @@
 """
 
 
-def fetch_data():
+def fetch_data(trusted_only=True):
     """Fetches the data from the Comparisons model
 
     Returns:
     - comparison_data: list of
         [   contributor_id: int, video_id_1: int, video_id_2: int,
             criteria: str, score: float, weight: float  ]
     """
+    comparisons_queryset = ComparisonCriteriaScore.objects.all().prefetch_related("comparison")
+    
+    if trusted_only:
+        comparisons_queryset = comparisons_queryset.filter(comparison__user__in=User.trusted_users())
     comparison_data = [
-        [
-            ccs.comparison.user_id,
-            ccs.comparison.entity_1_id,
-            ccs.comparison.entity_2_id,
-            ccs.criteria,
-            ccs.score,
-            ccs.weight,
-        ]
-        for ccs in ComparisonCriteriaScore.objects
-            .filter(comparison__user__in=User.trusted_users())
-            .prefetch_related("comparison")
+            [
+                ccs.comparison.user_id,
+                ccs.comparison.entity_1_id,
+                ccs.comparison.entity_2_id,
+                ccs.criteria,
+                ccs.score,
+                ccs.weight,
+            ]
+            for ccs in comparisons_queryset
     ]
+
     return comparison_data
 
 
-def save_data(video_scores, contributor_rating_scores):
+def save_data(video_scores, contributor_rating_scores, trusted_only=True):
     """
     Saves in the scores for Entities and ContributorRatings
     """
     default_poll_pk = Poll.default_poll_pk()
 
-    EntityCriteriaScore.objects.all().delete()
-    EntityCriteriaScore.objects.bulk_create(
-        [
-            EntityCriteriaScore(
-                entity_id=video_id,
-                criteria=criteria,
-                score=score,
-                uncertainty=uncertainty,
+    if trusted_only:
+        EntityCriteriaScore.objects.all().delete()
+        EntityCriteriaScore.objects.bulk_create(
+            [
+                EntityCriteriaScore(
+                    entity_id=video_id,
+                    criteria=criteria,
+                    score=score,
+                    uncertainty=uncertainty,
+                )
+                for video_id, criteria, score, uncertainty in video_scores
+            ]
+        )
+
+    if trusted_only:
+        trusted_contributor_rating_scores = set(
+            (contributor_id, video_id, criteria, score, uncertainty)
+            for (contributor_id, video_id, criteria, score, uncertainty) in contributor_rating_scores
+            if (contributor_id,) in User.trusted_users().values_list (
+                "id"
             )
```

Comment:
Filter to get only the `ContributorRatingScore` we whan to update.

Reply:
I took the liberty to make some adjustments, especially here to avoid redundant SQL queries. 
```python
            for (contributor_id, video_id, criteria, score, uncertainty) in contributor_rating_scores
            if (contributor_id,) in User.trusted_users().values_list (
                "id"
            )
```
would fetch the list trusted users for every score.